### ISSUE TITLE
Inset doesn't work inside subplot

### ISSUE THREAD

**Issue by seisman** [GenericMappingTools/gmt#1457]
Tried to rewrite the modern ex44 with subplot, but found that inset doesn't work in subplot.

The new script:
```
gmt begin ex44 ps
  gmt subplot begin 2x1 -Fs6i/4i -B -BWSne
    gmt coast -R10W/5E/35N/44N -JM6i -EES+gbisque -Gbrown -Wfaint -N1/1p -Sazure1 -Df --FORMAT_GEO_MAP=dddF -c0
    gmt inset begin -DjTR+w2i/0.93i+o0.15i/0.1i -F+gwhite+p1p+c0.1c+s
        gmt coast -R15W/35E/30N/48N -JM? -Da -Gbrown -B0 -EES+gbisque --MAP_FRAME_TYPE=plain
    gmt inset end

    gmt coast -R110E/170E/44S/9S -JM6i -Wfaint -N2/1p -EAU+gbisque -Gbrown -Sazure1 -Da --FORMAT_GEO_MAP=dddF -c1
    gmt inset begin -DjTR+w1.5i+o0.15i/0.1i -F+gwhite+p1p+c0.1c+s
        gmt coast -Rg -JG120/30S/? -Da -Gbrown -A5000 -Bg -Wfaint -EAU+gbisque
    gmt inset end
  gmt subplot end
gmt end
```

Error message:
```
inset [ERROR]: Unrecognized option -X
inset [ERROR]: Unrecognized option -Y
inset [ERROR]: Unrecognized option -X
inset [ERROR]: Unrecognized option -Y
inset [ERROR]: Unrecognized option -X
inset [ERROR]: Unrecognized option -Y
inset [ERROR]: Unrecognized option -X
inset [ERROR]: Unrecognized option -Y
```

**Comment by PaulWessel** [comment]
The horizontal shift of the major frames goes away if I plot Australia first.  So tied to that white gap shift.  Probably are placing one of those +dx, -dx translations inside a gsave/grestore and the other outside, so it does not revert properly.  Will look in the afternoon after class.

**Comment by PaulWessel** [comment]
A simplified version showing the bug:

```
gmt begin bug png
  gmt subplot begin 2x1 -Fs6i/4i -B -BWSne
    gmt basemap -R10W/5E/35N/44N -JM6i -c0
    gmt inset begin -DjTR+w2i/0.93i+o0.2i/0.1i -F+p1p,red
        gmt basemap -R15W/35E/30N/48N -JM? -B0 --MAP_FRAME_TYPE=plain
    gmt inset end

    gmt basemap -R110E/170E/44S/9S -JM6i -c1
    gmt inset begin -DjTR+w1.5i+o0.2i/0.1i -F+p1p,red
        gmt basemap -Rg -JG120/30S/? -Bg
    gmt inset end
  gmt subplot end
gmt end show
```
![bug](https://user-images.githubusercontent.com/26473567/63894465-66dcea80-c988-11e9-96e6-62f6980cc420.png)

The problems I see are:

1. The placement of the rectangle in the top subplot is wrong.  It should have the same offset relative to the top right corner as the second plot
2. The top inset plot (black frame) is shifted with respect to the inset (red frame - which is in the wrong location anyway)
3. A horizontal shift is added before the second (lower) map is plotted.

Agree that those are the problems here, @seisman ?

**Comment by seisman** [comment]
I think so.

**Comment by seisman** [comment]
Closed by #1461.
